src/util/algos/sort.c

```diff
@@ -0,0 +1,21 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+
+#include "mpiimpl.h"
+
+void MPIR_insertion_sort(int *arr, int n)
```

Comment:
This is a general routine. I think it belongs in `MPL`, probably in a `misc` folder. In general, MPL hosts utility routines that don't require context -- both usage context and implementation context -- to maintain.